client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,46 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to
+     * 'CONSUMPTION' based retention {@link RetentionPolicy.RetentionType#CONSUMPTION} ,
+     * the Reader Group needs to periodically notify StreamCuts to Controller to indicate
+     * the point in the Stream till which it has completed consuming data.
+     * This can be done by setting the retentionConfig in ReaderGroupConfig {@link ReaderGroupConfig}
+     * to 'CONSUMPTION_BASED_USER_STREAMCUT' or 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
```

Comment:
```suggestion
     * This can be done automatically at each checkpoint by setting the `retentionConfig` to {@link ReaderGroupConfig#CONSUMPTION_BASED_AT_LAST_CHECKPOINT}.
     * Alternatively the ReaderGroup can request the data be held until {@link  ... } is called by setting `retentionConfig` to {@link ReaderGroupConfig#CONSUMPTION_BASED_USER_STREAMCUT}.
```

Reply:
fixed